PWGDQ/Tasks/tableReader.cxx

```diff
@@ -766,68 +1026,94 @@ struct AnalysisSameEventPairing {
               fHistMan->FillHistClass(histNames[icut][2].Data(), VarManager::fgValues);
             }
           }
+          // additional histograms with prefiltering
+          if (fPrefilter) {
+            // By default take the bits from table maker, but uses the one newly calculated if we have the process function WithDalitzBits
+            for (int dalitzCut = 0; dalitzCut < nDalitzCuts; dalitzCut++) {
+              if (!((uint8_t(1) << dalitzCut) & twoTracksDalitzFilter)) { // check that none of the track is tagged as dalitz
+                // TrackCut_DalitzCuts histograms
+                if (t1.sign() * t2.sign() < 0) {
+                  fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][0].Data(), VarManager::fgValues);
+                } else {
+                  if (t1.sign() > 0) {
+                    fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][1].Data(), VarManager::fgValues);
+                  } else {
+                    fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][2].Data(), VarManager::fgValues);
+                  }
+                }
+              } // end if not dalitz-rejected
+            }   // end loop dalitz cuts
+          }     // end if prefilter
         } // end if (filter bits)
       }   // end for (cuts)
     }     // end loop over pairs
   }
 
+  void processJpsiToEESkimmedDalitzBits(soa::Filtered<MyEventsVtxCovSelected>::iterator const& event, soa::Filtered<MyBarrelTracksSelected> const& tracks, DalitzBitsReduced const& dalitzBits)
+  {
+    // Reset the fValues array
+    VarManager::ResetValues(0, VarManager::kNVars);
+    VarManager::FillEvent<gkEventFillMap>(event, VarManager::fgValues);
+    runSameEventPairing<VarManager::kJpsiToEE, gkEventFillMap, gkTrackFillMapDalitzBits>(event, tracks, tracks, dalitzBits);
+  }
```

Comment:
I am not sure we need a separate process function for this. Firstly: if we use the Dalitz bits from the skimming, then that is just a cut we specify via CutsLibrary (the info is in the filterTag of the track, no need of any new table).
Secondly: If we do prefiltering with the task, then in the config json we can switch it on, and in the same event pairing we can just have a configurable that specifies to apply or not the prefilter decision from the prefilter task. So really no need for a new process function, which is good, because the functionality is then applied directly to all existing process functions, which is highly preferable.